src/main.ts

```diff
@@ -8,38 +8,178 @@ import {
 	normalizePath,
 	TextComponent,
 	ExtraButtonComponent,
+	ToggleComponent,
 } from 'obsidian';
 
+import {Mutex} from 'async-mutex';
+
 import { MARKWHEN_ICON } from './icons';
 import { VIEW_TYPE_MARKWHEN, MarkwhenView } from './MarkwhenView';
 
+import * as fs from 'fs';
+import * as path from 'path';
+
 interface MarkwhenPluginSettings {
 	folder: string;
+	deftoselectedtoggle: boolean;
```

Comment:
I don't know what this variable name means - it should be camel cased

Reply:
This variable is to hold the boolean value for "open a new file in current folder (selected/opened file's folder) || open a file in folder with name given in settings"